src/particles/pusher/FieldGather.H

```diff
@@ -241,23 +241,37 @@ void doLaserGatherShapeN (const amrex::ParticleReal xp,
 
     // Gather field Aabssq, AabssqDxp, and AabssqDxp on particle from field on grid a_arr
     // the derivative is calculated on the fly
-    // TODO: complex conjugate will be needed when a becomes complex (required for laser evolution)
     for (int iz=0; iz<=depos_order_z; iz++){
         for (int iy=0; iy<=depos_order_xy; iy++){
             for (int ix=0; ix<=depos_order_xy; ix++){
-                Aabssqp += sx_cell[ix]*sy_cell[iy]*sz_cell[iz]*
-                    a_arr(j_cell+ix  , k_cell+iy  , amrex::lbound(a_arr).z)*
-                    a_arr(j_cell+ix  , k_cell+iy  , amrex::lbound(a_arr).z);
-                AabssqDxp += sx_cell[ix]*sy_cell[iy]*sz_cell[iz]/(2.0_rt*dx[0])*
-                  ( a_arr(j_cell+ix+1, k_cell+iy  , amrex::lbound(a_arr).z)*
-                    a_arr(j_cell+ix+1, k_cell+iy  , amrex::lbound(a_arr).z)
-                   -a_arr(j_cell+ix-1, k_cell+iy  , amrex::lbound(a_arr).z)*
-                    a_arr(j_cell+ix-1, k_cell+iy  , amrex::lbound(a_arr).z));
-                AabssqDyp += sx_cell[ix]*sy_cell[iy]*sz_cell[iz]/(2.0_rt*dx[1])*
-                  ( a_arr(j_cell+ix  , k_cell+iy+1, amrex::lbound(a_arr).z)*
-                    a_arr(j_cell+ix  , k_cell+iy+1, amrex::lbound(a_arr).z)
-                   -a_arr(j_cell+ix  , k_cell+iy-1, amrex::lbound(a_arr).z)*
-                    a_arr(j_cell+ix  , k_cell+iy-1, amrex::lbound(a_arr).z));
+                const amrex::Real x00y00 =
+                    a_arr(j_cell+ix  , k_cell+iy  , amrex::lbound(a_arr).z,0)*
+                    a_arr(j_cell+ix  , k_cell+iy  , amrex::lbound(a_arr).z,0)+
+                    a_arr(j_cell+ix  , k_cell+iy  , amrex::lbound(a_arr).z,1)*
+                    a_arr(j_cell+ix  , k_cell+iy  , amrex::lbound(a_arr).z,1);
```

Comment:
I think a `abs(real, imag)` function would be good here.

Reply:
Thanks, that's included now